src/directives/schemaDirectiveVisitors.ts

```diff
@@ -37,27 +36,25 @@ export class HasRoleDirective extends SchemaDirectiveVisitor {
 
   public visitFieldDefinition (field: any) {
     const { resolve = defaultFieldResolver } = field
-    const { error, value } = this.validateArgs()
-    if (error) {
-      throw error
-    }
-
-    const { roles } = value
-
+    const roles = this.validateArgs(this.args)
     field.resolve = hasRole(roles)(resolve)
   }
 
-  public validateArgs () {
-    // joi is dope. Read the docs and discover the magic.
-    // https://github.com/hapijs/joi/blob/master/API.md
-    const argsSchema = Joi.object({
-      role: Joi.array().required().items(Joi.string()).single()
-    })
-
-    const result = argsSchema.validate(this.args)
-
-    // result.value.role will be an array so it makes sense to add the roles alias
-    result.value.roles = result.value.role
-    return result
+  // validate a potential string or array of values
+  // if an array is provided, cast all values to strings
+  public validateArgs (args: {[name: string]: any}): Array<string> {
```

Comment:
I would expect to validate the function to work with the immutable objects. 
Here we transform arguments into new structure

```suggestion
  public transformArgs (args: {[name: string]: any}): Array<string> {
```